portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -19,14 +19,28 @@
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import java.util.*; 
+import com.google.gson.Gson;
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private final ArrayList<String> list = new ArrayList<String>();
+
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    fillList(list);
+    Gson gson = new Gson();
+    String json = gson.toJson(list);
+
+    response.setContentType("application/json");
+    response.getWriter().println(json);
+  }
+
+  public void fillList(ArrayList<String> list){
```

Comment:
It often makes debugging easier if helper functions don't modify the variables they're passed. Do you think that would make sense here?

In other words, instead of calling "fillList(list)", you would have a method, maybe "filledList()", which creates the list, populates it, and returns it. Then in "doGet()", you would have "list = filledList()". Does that make sense? What do you think?

I'm also happy to expand on why this often makes debugging easier, if that would be helpful.

Reply:
I think I understand why it would make debugging easier. Is it because you can modularize the code and debug the function as a unit? Thus, when debugging, you can focus on what the function is doing wrong and not worry about malformed inputs? Additionally, when calling "list = filledList()", any errors that come from this line would be from the function and not from a bad input? Thank you!

Reply:
Yep, exactly. Less that can go wrong :)

Reply:
I made the adjustments to this branch, thanks!